File: WebApi/WebApi/Models/Gift.cs
namespace WebApi.Models
{
    public class Gift
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; }=string.Empty;
        public List<Purchase> Purchases { get; set; } = new List<Purchase>();
        public int CategoryId { get; set; }
        public Category Category { get; set; }
        public int DonorId { get; set; }
        public Donor Donor { get; set; }
        public Decimal PriceCard { get; set; }
        public int Quantity { get; set; }
        public static bool IsRandom { get; set; } = false;


    }
}

File: Angular/Angular/src/Model/Gift.ts
import { Category } from "./Category";
import { Donor } from "./Donor";
import { Purchase } from "./Purchase";
import { User } from "./User";

export class Gift {
    public id: number = 0;
    public name: string = '';
    public description: string = '';
    public purchases: Purchase[] = [];
    public categoryId: number = 0;
    public category: Category = new Category();
    public donorId: number = 0;
    public donor: Donor = new Donor();
    public priceCard: number = 0;
    public quantity: number = 0;
    public static isRandom: boolean = false;
    categoryName?: string;
    pictureId: number = 0;
    public noImage?: boolean;
    public winner: User = new User();
    public winnerName?: string; 
    searchName: string = '';

}
